gen/calc.py

```diff
@@ -711,6 +711,76 @@ def merged_dict(dict_a, dict_b):
     return yaml.dump(json.dumps(merged_checks, indent=2))
 
 
+def calculate__superuser_credentials_given(
```

Comment:
We should add unit tests for new gen parameters: `superuser_service_account_uid` and `superuser_service_account_public_key`, especially validation of wrong values provided.

Example of existing tests: https://github.com/dcos/dcos/blob/186e38c3279942a83e6396355d6666671b62e4d0/gen/tests/test_adminrouter_tls_conf.py

Reply:
Would you be OK with noting this as a follow-up task? To me it is not yet 100 % clear how the unit tests will cover something different from the E2E test work tracked via https://jira.mesosphere.com/browse/DCOS-45325.

Reply:
The E2E tests are "expensive" in terms of time and infrastructure and they are testing that new parameters are working end-to-end, across different components. The unit tests should test also validation errors and they confirm that changes specifically in `gen` package are working as expected. Its also a good reference for what values can be provided to newly introduced parameters.

Reply:
Good discussion. I have extended https://jira.mesosphere.com/browse/DCOS-45325 to also track adding unit tests for `gen` for these two parameters.

Reply:
As far as @timaa2k and I can tell, this code is never run. That is, we cannot make a test which shows the error.